lib/Net/Kubernetes/Role/ResourceCatalog.pm

```diff
@@ -0,0 +1,76 @@
+package Net::Kubernetes::Role::ResourceCatalog;
+
+use Moose::Role;
+
+requires 'server_version';
+requires 'url';
+
+my %endpoint_catalog = (
+	1.5 => {
```

Comment:
I've tested against 1.2 and 1.5. They both seem to work for resources I'd expect to see in each version. The default behavior is what what expected as of 1.2. The problem is that I can't find documentation for the intermediate versions. 

The following documentation only starts at 1.5:
https://kubernetes.io/docs/reference/